src/core/bluetoothreceiver.cpp

```diff
@@ -40,6 +40,12 @@ BluetoothReceiver::BluetoothReceiver( QObject *parent ) : QObject( parent ),
   }
 }
 
+BluetoothReceiver::~BluetoothReceiver()
+{
+  mSocket.release(); // deleted by QgsGpsConnection::cleanupSource
```

Comment:
Could we use a raw pointer instead of a unique_ptr?
If we don't hand over ownership, we shouldn't be using unique_ptr?

Reply:
That'd be perfectly fine yes

Reply:
@m-kuhn , done.